native/src/main/java/io/ballerina/lib/avro/serialize/NullSerializer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.avro.serialize;
+
+import io.ballerina.lib.avro.serialize.visitor.SerializeVisitor;
+
+public class NullSerializer extends Serializer {
+
+    @Override
+    public Object convert(SerializeVisitor serializeVisitor, Object data) throws Exception {
+        if (data != null) {
+            throw new Exception("The value does not match with the null schema");
+        }
+        return null;
+    }
+}
```

Comment:
Do we really need a separate class to handle null values?

Reply:
When serializing, a specific subclass of `Serializer` is generated for each schema type. If we skip making a separate subclass just for the `Null` schema type, we'll have to come up with a different method to handle them, disrupting the polymorphism used here.